python/lsst/ctrl/execute/slurmPlugin.py

```diff
@@ -162,12 +151,10 @@ def submit(self, platform, platformPkgDir):
 
         # run the sbatch command
         template = Template(self.getLocalScratchDirectory())
-        localScratchDir = template.substitute(USER_SCRATCH=self.getUserScratch())
-        slurmSubmitDir = os.path.join(localScratchDir, self.defaults["DATE_STRING"])
-        if not os.path.exists(localScratchDir):
-            os.mkdir(localScratchDir)
-        if not os.path.exists(slurmSubmitDir):
-            os.mkdir(slurmSubmitDir)
+        localScratchDir = Path(template.substitute(USER_SCRATCH=self.getUserScratch()))
```

Comment:
Also have the option of using `ResourcePath` here if you don't want to mix Path with ResourcePath.

```python
localScratchDir = ResourcePath(template.substitute(USER_SCRATCH=self.getUserScratch()), forceDirectory=True)
slurmSubmitDir = localScratchDir.join(self.defaults["DATE_STRING"], forceDirectory=True)
localScratchDir.mkdir()
slurmSubmitDir.mkdir()
```

Reply:
That's good to know, but `ResourcePath` doesn't support the handy `__div__` overload used to create new paths (`Path(".") / "some" / "other" / "directory"`)

Reply:
Because no-one has asked for it. It's easy to add. File a Jira ticket.